mailpoet/assets/js/src/automation/integrations/mailpoet/steps/send_email/edit/design_email_button.tsx

```diff
@@ -49,7 +49,7 @@ export function DesignEmailButton(): JSX.Element {
   // and the workflow is saved, we can safely redirect to the email design flow.
   useEffect(() => {
     if (emailId && workflowSaved) {
```

Comment:
This might be more of a comment for @JanJakes because I believe he [wrote the original code](https://github.com/mailpoet/mailpoet/commit/8319c1e4b676c5e6399d16f0281d08e526c19a93#diff-8e2de473afa9b9ebdcbd86a22b1edcdb5352f6f289da587ea1363d685d3d4326R14). `workflowSaved` is always going to be true because it's actually getting the currently selected step.

```js
  const { selectedStep, workflowId, workflowSaved } = useSelect(
    (select) => ({
      selectedStep: select(storeName).getSelectedStep(),
      workflowId: select(storeName).getWorkflowData().id,
      workflowSaved: select(storeName).getSelectedStep(),
    }),
    [],
  );
```

When I change the definition to the actual `workflowSaved` state property, the redirect stops working.

I think this redirection behavior might currently depend on the fact that this effect fires again when the step (aka `workflowSaved` right now) changes to have an `email_id` defined in its `args`. In any case, when I try changing it to actually reference the `workflowSaved` data, I never see this component re-render with `workflowSaved` === `true`.

I don't know if it's related, but sometimes this redirect doesn't work for me. It seems to require something like the following:

1. Create a workflow with more than 1 email step
2. Create an email for one of the steps
3. Return to the workflow editor
4. Select the step that has an email associated with it (necessary step!)
5. Select a different email step without an email yet
6. Click the button to create an email
7. Observe that the redirect is not triggered

Reply:
Thanks for pointing that out. I had a closer look. The problem is, the component would not render when an `email_id` is present. I am not quite sure, why the redirect happened at all, but it was not reliable. I have reworked the component and I think it works reliably now. [dd22635](https://github.com/mailpoet/mailpoet/pull/4391/commits/dd226354c2b6db36a5ba2a32869ec36743dfe150)